README.md

```diff
@@ -2,92 +2,707 @@
+```
+
+### Writing Objects
+
+To upload local files as objects, you need proper authorization (see [Authorization](#authorization) section):
+
+```python
+import fsspec
+
+# Method 1: Using fsspec.put_file with schemes (recommended)
+# Note: Pass storage_options for authorization
+fsspec.put_file('/local/path/file.txt',
+                'osdf:///namespace/remote/path/object.txt',
+                storage_options={"headers": {"Authorization": "Bearer YOUR_TOKEN"}})
+
+# Upload multiple local files as objects
+fsspec.put('/local/directory/',
+           'osdf:///namespace/remote/path/',
+           recursive=True,
+           storage_options={"headers": {"Authorization": "Bearer YOUR_TOKEN"}})
+
+# Method 2: Using PelicanFileSystem directly
+from pelicanfs.core import PelicanFileSystem
+pelfs = PelicanFileSystem("pelican://osg-htc.org",
+                          headers={"Authorization": "Bearer YOUR_TOKEN"})
+pelfs.put('/local/path/file.txt', '/namespace/remote/path/object.txt')
+```
+
+### Downloading Objects
+
+```python
+import fsspec
+
+# Method 1: Using fsspec.get with schemes (recommended)
+# Download an object to a local file
+fsspec.get('osdf:///ospool/uc-shared/public/object.txt', '/local/path/file.txt')
+
+# Download multiple objects
+fsspec.get('osdf:///ospool/uc-shared/public/', '/local/directory/', recursive=True)
+
+# Method 2: Using PelicanFileSystem directly
+from pelicanfs.core import PelicanFileSystem
+pelfs = PelicanFileSystem("pelican://osg-htc.org")
+pelfs.get('/ospool/uc-shared/public/object.txt', '/local/path/file.txt')
+```
 
-The following describes how to specify endpoints to get data from, rather than letting PelicanFS and the director determine the best cache. PelicanFS allows you to specify whether to read directly from the origin (bypassing data staging altogether) or to name a specific cache to stage data into.
+## Advanced Configuration
+
+### Specifying Endpoints
 
-**Note**
-> If both direct reads and a specific cache are set, PelicanFS will use the specified cache and ignore the direct reads setting.
+PelicanFS allows you to control where data is read from, rather than letting the director automatically select the best cache.
 
+**Note:** The `direct_reads` and `preferred_caches` settings are mutually exclusive. If `direct_reads=True`, data will always be read from origins and `preferred_caches` will be ignored. If `direct_reads=False` (the default), then `preferred_caches` will be used if specified.
 
 #### Enabling Direct Reads
 
-Sometimes you might wish to read data directly from an origin rather than via a cache. To enable this at PelicanFileSystem creation, just pass in `direct_reads=True` to the constructor.
+Read data directly from origins, bypassing caches entirely:
 
 ```python
 pelfs = PelicanFileSystem("pelican://osg-htc.org", direct_reads=True)
 ```
 
-#### Specifying a Cache
+This is useful when:
+- You're close to the origin server
+- You want to ensure you're reading the most up-to-date data
```

Comment:
I get what's meant by this, but since we call objects in Pelican immutable, it doesn't really make sense to refer to "up-to-date data".